src/main/java/com/squareup/javapoet/Util.java

```diff
@@ -82,4 +154,23 @@ public static String join(String separator, List<String> parts) {
     result.addAll(b);
     return result;
   }
+
+  public static Set<Modifier> getMissingImplicitModifiers(Set<Modifier> modifiers,
```

Comment:
Can you explain what `requireSubsetOf` should be doing? It's not exactly clear from the name, as that would contain some invalid combos as currently specified.